internal/plugin.go

```diff
@@ -648,6 +651,31 @@ func (p *Plugin) assertValidConfig() error {
 		)
 	}
 
+	// validate placement and binding names are DNS compliant
+	placementName := ""
+	if p.PolicyDefaults.Placement.PlacementRuleName != "" {
```

Comment:
Could you please just validate all these values if they are set so this code doesn't need to be aware of the order of priority?